specs/Standards/Sould Bound Token/README.md

```diff
@@ -0,0 +1,235 @@
+interface SbtRevokeLog {
+  owner: string;
+  tokens: string[];
+  memo?: string;
+}
+
+// An event emitted when a recovery process succeeded to reassign SBT.
+// Arguments
+// * `old_owner`: "old_account.near"
+// * `new_owner`: "new_account.near"
+// * `token_ids`: ["1", "abc"]
+// * `memo`: optional message
+interface SbtRecoverLog {
+  old_owner: string;
+  new_owner: string;
+  tokens: string[];
+  memo?: string;
+}
+
+// An event emitted when a existing tokens are renewed.
+// Arguments
+// * `tokens`: ["1", "abc"]
+// * `memo`: optional message
+interface SbtRenewLog {
+  tokens: uint64[];
+  memo?: string;
+}
+```
+
+Whenever a recovery is made in a way that an existing SBT is burned, the `SbtRevokeLog` event must be emitted.
+
+```typescript
+class SbtBurnedAccounts {
+  burn(account_id) {
+    this.burned_accounts[this.account_id][this.predecessor_id] = true;
+  }
+
+  // query
+  is_burned(account_id, sbt_token_contract): bool {
+    return this.burned_accounts[account_id][sbt_token_id];
+  }
+}
+```
+
+### Recommended functions
+
+Although the functions below are not part of the standard (depending on a use case, they may need different parameters), we recommend them as a part of implementation and we also provide them in the reference implementation.
+These functions should emit appropriate events.
+
+```typescript=
+
+interface SBT {
+
+    /// Function for recovery committee, which can be either
+    /// the issuer, DAO, or an operator smart contract, authorized for the
+    /// recovery process.
+    /// Emits `SbtRecoverLog` when the recover process succeeds.
+    /// Returns an error if the recovery process failed.
+    recover(token_id: uint64, old_address: string, new_address: string): error | undefined;
+
+    /// Function to revoke and burn SBT. Must emit `SbtRevokeLog` event.
+    /// Return true if a token_id is an valid, active SBT. Otherwise returns false.
+    revoke(token_id: uint64): bool;
+}
+```
+
+## Reference Implementation
+
+- https://github.com/alpha-fi/i-am-human/tree/master/contracts/soulbound
+
+## Example Flow
+
+## Consequences
+
+### Positive
+
+- Template and set of guidelines for creating SBT tokens.
+- Ability to create SBT aggregators.
+- Ability to use SBT as a primitive to model non KYC identity, badges, certificates etc...
+- SBT can be further used for "lego" protocols, like: Proof of Humanity (dicussed for NDC Governance), undercollateralized lending, role based authentication sytems, innovative economic and social applications...
+- Stanarized recoverability mechanism.
+- SBT are considered as a basic primitive for Decentralized Societies.
+- new way to implement Sybil attack resistance.
+
+### Netural
+
+- API follows the NEP-171 (NFT) standard.
+  NOTE: we can decide to use `nft_` prefix whenever possible.
+
+### Negative
+
+- new set of events to be handled by the indexer. However this is not an issue if we decide to use subset of events used by the NEP-171 (NFT).
```

Comment:
Same reasoning as above. New events. Updating indexers is not as bad as tokens being misclassified.

Reply:
Agree. But it's negative from the indexer perspective: new support will have to be added.